privmx-endpoint/src/main/cpp/modules/ThreadApi.cpp

```diff
@@ -471,59 +471,117 @@ Java_com_simplito_java_privmx_1endpoint_modules_thread_ThreadApi_updateMessage(
+                }
+
+                auto subscription_ids_c = getThreadApi(ctx, thiz)->subscribeFor(
+                        subscription_queries_c);
+
+                jobject javaArrayList = env->NewObject(arrayListCls, initMID);
+                for (auto &id_str: subscription_ids_c) {
+                    jstring java_id_str = ctx->NewStringUTF(id_str.c_str());
+                    env->CallBooleanMethod(javaArrayList, addToListMID, java_id_str);
+                }
+
+                return javaArrayList;
+            }
+    );
+
+    if (ctx->ExceptionCheck()) {
+        return nullptr;
+    }
+    return result;
 }
+
 extern "C"
 JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_thread_ThreadApi_subscribeForMessageEvents(
+Java_com_simplito_java_privmx_1endpoint_modules_thread_ThreadApi_unsubscribeFrom(
         JNIEnv *env,
         jobject thiz,
-        jstring thread_id
+        jobject subscription_ids
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(thread_id, "Thread ID")) {
+    if (ctx.nullCheck(subscription_ids, "Subscription IDs")) {
         return;
     }
-    ctx.callVoidEndpointApi([&ctx, &thiz, &thread_id]() {
-        getThreadApi(ctx, thiz)->subscribeForMessageEvents(
-                ctx.jString2string(thread_id)
-        );
+
+    ctx.callVoidEndpointApi([&ctx, &env, &thiz, &subscription_ids]() {
+        auto subscription_ids_arr = ctx.jObject2jArray(subscription_ids);
+        auto subscription_ids_c = std::vector<std::string>();
+
+        int length = ctx->GetArrayLength(subscription_ids_arr);
+        for (int i = 0; i < length; i++) {
+            jobject arrayElement = ctx->GetObjectArrayElement(subscription_ids_arr, i);
+            subscription_ids_c.push_back(ctx.jString2string((jstring) arrayElement));
+        }
+
+        getThreadApi(ctx, thiz)->unsubscribeFrom(subscription_ids_c);
     });
 }
+
+
 extern "C"
-JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_thread_ThreadApi_unsubscribeFromMessageEvents(
+JNIEXPORT jstring JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_thread_ThreadApi_buildSubscriptionQuery(
         JNIEnv *env,
         jobject thiz,
-        jstring thread_id
+        jlong eventType,
+        jlong selectorType,
+        jstring selectorId
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(thread_id, "Thread ID")) {
-        return;
+    if (ctx.nullCheck(selectorId, "SelectorID")) {
+        return nullptr;
     }
-    ctx.callVoidEndpointApi([&ctx, &thiz, &thread_id]() {
-        getThreadApi(ctx, thiz)->unsubscribeFromMessageEvents(
-                ctx.jString2string(thread_id)
-        );
-    });
+
+    jstring result;
+    ctx.callResultEndpointApi<jstring>(
+            &result,
+            [&ctx, &env, &thiz, &eventType, &selectorType, &selectorId]() {
+                auto c_eventType = parseThreadEventType(ctx, (long) eventType);
```

Comment:
you can easy cast long value to enum like this
```suggestion
                auto c_eventType = static_cast<thread::EventType>(eventType);
```